test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -1248,3 +1252,51 @@ UCS_TEST_P(test_ucp_mmap_export, export_import) {
 }
 
 UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(test_ucp_mmap_export)
+
+#if HAVE_CUDA
+class test_ucp_mmap_mgpu : public ucs::test {
+};
+
+UCS_TEST_F(test_ucp_mmap_mgpu, switch_gpu) {
```

Comment:
do we need to add tests for transfer cuda_copy or cuda_ipc, mem type cuda or vmm/mallocasync:
- buf1 on device1, buf2 on device2, copy happening under progress with context of unrelated device3

Reply:
Agree, we will need more test cases.
The test in this PR is the simplest one. And it fails. Once it is fixed, we will add more scenarios.